src/Test/L0/Container/ContainerInfoL0.cs

```diff
@@ -1,38 +1,64 @@
+            private string Title;
+
+            public MountVolumeTest(string input, MountVolume expected, string title="")
+            {
+                this.Input = input;
+                this.Expected = expected;
+                this.Title = title;
+            }
+
+            public void run()
+            {
+                MountVolume got = new MountVolume(Input);
+                Assert.True(Expected.SourceVolumePath == got.SourceVolumePath, $"{Title} - testing property SourceVolumePath. Expected: '{Expected.SourceVolumePath}' Got: '{got.SourceVolumePath}' ");
+                Assert.True(Expected.TargetVolumePath == got.TargetVolumePath, $"{Title} - testing property TargetVolumePath. Expected: '{Expected.TargetVolumePath}' Got: '{got.TargetVolumePath}'");
+                Assert.True(Expected.ReadOnly == got.ReadOnly, $"{Title} - testing property ReadOnly. Expected: '{Expected.ReadOnly}' Got: '{got.ReadOnly}'");
+            }
+        }
+
         [Fact]
         [Trait("Level", "L0")]
         [Trait("Category", "Worker")]
         public void MountVolumeConstructorParsesStringInput()
         {
-            // Arrange
-            MountVolume target = new MountVolume("/dst/dir"); // Maps anonymous Docker volume into target dir
-            MountVolume source_target = new MountVolume("/src/dir:/dst/dir"); // Maps source to target dir
-            MountVolume target_ro = new MountVolume("/dst/dir:ro");
-            MountVolume source_target_ro = new MountVolume("/src/dir:/dst/dir:ro");
-
-            // Assert
-            Assert.Null(target.SourceVolumePath);
-            Assert.Equal("/dst/dir", target.TargetVolumePath);
-            Assert.False(target.ReadOnly);
+            List<MountVolumeTest> tests = new List<MountVolumeTest>
+            {
+                // Unix style paths
+                new MountVolumeTest("/dst/dir", new MountVolume(null, "/dst/dir", false), "Maps anonymous Docker volume into target dir"), 
+                new MountVolumeTest("/src/dir:/dst/dir", new MountVolume("/src/dir", "/dst/dir", false), "Maps source to target dir"), 
+                new MountVolumeTest("/dst/dir:ro", new MountVolume(null, "/dst/dir", true), "Maps anonymous Docker volume read-only into target dir"), 
+                new MountVolumeTest("/src/dir:/dst/dir:ro", new MountVolume("/src/dir", "/dst/dir", true), "Maps source to read-only target dir"), 
+                new MountVolumeTest(":/dst/dir", new MountVolume(null, "/dst/dir", false), "Maps anonymous Docker volume into target dir with leading colon"), 
+                new MountVolumeTest("/c/src/dir:/c/dst/dir", new MountVolume("/c/src/dir", "/c/dst/dir", false), "Maps source to target dir prefixed with /c/"), 
+                new MountVolumeTest("/src/dir\\:with\\:escaped\\:colons:/dst/dir\\:with\\:escaped\\:colons", new MountVolume("/src/dir:with:escaped:colons", "/dst/dir:with:escaped:colons", false), "Maps source to target dir prefixed with escaped colons"), 
```

Comment:
On second thought, I'm not sure if escaping is the correct solution, unless its optional. We dont wont to force people to use a different syntax than they would just ordinarily type on the cli when running docker. And as far as I can tell, that type of syntax is not valid (do you have an example?) Moreover, if its optional, when would it be useful?

Reply:
The escaping is definitely optional. I automatically escape the colon after a drive letter. I added tests for explicit escaping in case someone needed to mount a directory with a colon in the name of a directory.